### PR TITLE
Track part ID moves during campaign loads that may affect refits

### PR DESCRIPTION
**Pull request by sixlettervariables**
This is an experimental change that may fix the tough to track down part ID changes affecting refits at CPNX load #1193 #1180 #1138 #1123 (to name a few).

*note: best compared with whitespace ignored*

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/Campaign.java [hunk 1]
```diff
@@ -1425,23 +1426,26 @@ public void addPart(Part p, int transitDays) {
                         ((Armor) spare).setAmount(((Armor) spare).getAmount()
                                 + ((Armor) p).getAmount());
                         MekHQ.triggerEvent(new PartChangedEvent(spare));
+                        p.setId(-1);
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/Campaign.java]: What does the -1 part ID signify?

File: MekHQ/src/mekhq/campaign/Campaign.java [hunk 2]
```diff
@@ -1516,30 +1520,73 @@ public void addPartWithoutId(Part p) {
...
+                            }
+                        }
+                        ids = r.getNewUnitPartIds();
+                        for (int ii = 0; ii < ids.size(); ++ii) {
+                            int nid = (int)ids.get(ii);
+                            if (p.getId() == nid) {
+                                MekHQ.getLogger().info(Campaign.class, "addPartWithoutId",
+                                String.format("%s (%d) was merged with %s (%d) used in the new unit in a refit for %s", p.getName(),
+                                    p.getId(), mergedWith.getName(), mergedWith.getId(), u.getName()));
+                                ids.set(ii, mergedWith.getId());
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/Campaign.java]: What's the point of adding elements to the ids collection if you're not using it? Is it just a debugging thing?

File: MekHQ/src/mekhq/campaign/Campaign.java (hunk 1 above)
Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/Campaign.java]: Code which writes parts in some areas uses a full part description if the ID is less than or equal to zero. This is an attempt to force that if that code is the cause of the refit bugs.

File: MekHQ/src/mekhq/campaign/Campaign.java (hunk 2 above)
Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/Campaign.java]: Those are loaded at CPNX read and eventually get out of sync. I'm not sure if these IDs getting out of sync are the proximate cause, but it seemed worth doing if we can.

File: MekHQ/src/mekhq/campaign/Campaign.java (hunk 1 above)
Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/Campaign.java]: (The full description can't have an ID get out of sync, because it creates a brand new part)

**Pull request merged**